### ISSUE TITLE
TypeScript typings when using registerMethods

### ISSUE THREAD

**Issue by jesperorb** [facebook/jscodeshift#459]
Hello,

I am trying to use TypeScript in combination with `jscodeshift`. When using `Collection.registerMethods` how am I supposed to get the correct typings on `Collection`?

The methods gets registered correctly but my custom methods are not part of the types forcing me to manually cast when using any method if I want to use them. Below is a screenshot of when I have tried to register the methods.
<img width="1060" alt="Screenshot 2021-10-17 at 17 42 11" src="https://user-images.githubusercontent.com/21122051/137634793-3c72b23d-3d29-487b-90f7-b50a8a974227.png">

I have experimented some with using different [Declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) but can't seem to extend the types correctly.

Here is the repo I am trying to implement this in: https://github.com/jesperorb/jscodeshift-typescript

The file [`methods.ts`](https://github.com/jesperorb/jscodeshift-typescript/blob/main/src/methods.ts#L12-L34) contains the methods I want to register.

**Comment by cassels** [comment]
Hey @jesperorb, did you find a workaround? If not, I managed to get this to work through declaration merging as you suggested so posting here in case others stumble on this issue while searching for a solution (like I did). 

Because `Collection` is a `type` in the exported `core` namespace it's not possible to merge directly. However, I did have success merging the inner `Collection.Collection` interface that has unblocked me.

Try adding this to top of your `methods.ts` file.

```ts
declare module 'jscodeshift/src/Collection' {
  interface Collection<N> {
    findJSXElementProperty: typeof customCollectionMethods.findJSXElementProperty;
    findObjectPropertiesByName: typeof customCollectionMethods.findObjectPropertiesByName;
  }
}
```

Hope it helps 👍
